src/Core/Form/IdentifiableObject/Builder/FormBuilder.php

```diff
@@ -89,18 +98,32 @@ public function __construct(
      */
     public function getForm(array $data = [], array $options = [])
     {
-        if (is_array($defaultData = $this->dataProvider->getDefaultData())) {
+        // Fetch default data
+        $defaultData = $this->dataProvider->getDefaultData();
+        // Fetch initial options
+        $initialOptions = $options;
+
+        $this->hookDispatcher->dispatchWithParameters(
+            'action' . Container::camelize($this->registry->getType($this->formType)->getBlockPrefix()) . 'FormDataProviderDefaultData',
```

Comment:
Maybe the caramelize whole thing can be moved into a private method? Or at least this part `$this->registry->getType($this->formType)->getBlockPrefix()`

Reply:
Fixed